src/library/scala/collection/immutable/Vector.scala

```diff
@@ -659,6 +659,16 @@ class VectorIterator[+A](_startIndex: Int, endIndex: Int)
     }
   }
 
+  override def take(n: Int): Iterator[A] = {
+    val dropR = remainingElementCount - n
+    if(dropR > 0) {
```

Comment:
IMHO the following condition better expresses that this does no change unless "we take less than we have":  
```suggestion
    if(n < remainingElementCount) {
```
On the other hand, I am not sure if this would fully work with a negative  `n`. Further `take` operations would simply not change the state, still, do we want to leave the iterator on a dirty state, with `_endIndex_ < _startIndex`?

Reply:
Right, this needs to be guarded against a negative n.

I originally had `n < remainingElementCount` but the form in which it is needed afterwards is `remainingElementCount - n`. I think it also reads nicely: "If there is something to drop from the right, drop it".

`endIndex < startIndex` should be fine. I didn't investigate if `Vector` itself can deal with it, which it would if constructed via `VectorIterator#remainingVector` (which appears to be unused -- maybe it's used by the parallel collections) but the code in `VectorIterator` can. `remainingVector` should either be removed or changed to return `Vector.empty` in this case.